### ISSUE TITLE
When sim is set, simOptions should be set to default

### ISSUE THREAD

**Issue by jkomoros** [jkomoros/CASsim#49]
Currently it's set to default, but within Simulator, so option-control doesn't render it, which makes it a pain to have to go in an add the optional/default properties and tweak those.

Instead, setSimPropertyInConfig should set it to DEFAULT_SENTINEL

**Comment by jkomoros** [comment]
... this is surprisingly nontrivial, because it has circular references between simulation.js and options.js.

Effectively right now we're late-binding that default value, but now we have to early-bind it earlier in the workflow.

**Comment by jkomoros** [comment]
OK, new plan.  Factor out modfifiedConfigData to have the part where it does defaulting to be a function in the simulation.js package.

Simulation.cloneWithConfig grows simulation.cloneWithModification(path, value) which uses that sims config value, and can use the modifiedConfigData function

**Comment by jkomoros** [comment]
- [ ] Make sure the behavior actually does set the defaults in a visible way...

**Comment by jkomoros** [comment]
I was just confused with how this works, if it has to load a new simulator. But then I remembered that we set the sim value, and then immedaitely try to set the deafult, but it fails, but that's OK, because the machinery notes that there's now another sim necessary and that isn't loaded. Then when it loads, the modification is run again and now the default is there.

So this does work, and should work even for simulators that aren't yet loaded.
